internal/featureflags/featureflags.go

```diff
@@ -0,0 +1,85 @@
+/*
+ * @license
+ * Copyright 2023 Dynatrace LLC
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package featureflags
+
+import "os"
+
+// FeatureFlag represents a command line switch to turn certain features
+// ON or OFF. Values are read from environment variables defined by
+// the feature flag. The feature flag can have default values that are used
+// when the resp. environment variable does not exist
+type FeatureFlag struct {
+	// envName is the environment variable name
+	// that is used to read the value from
+	envName string
+	// defaultEnabled states whether this feature flag
+	// is enabled or disabled by default
+	defaultEnabled bool
+}
+
+// New creates a new FeatureFlag
+// envName is the environment variable the feature flag is loading the values from when evaluated
+// defaultEnabled defines whether the feature flag is enabled or not by default
+func New(envName string, defaultEnabled bool) FeatureFlag {
```

Comment:
With this is possible in code to do something like:
``` go
featureflags.New("MONACO_FEAT_NEW", false).Enable()
```
That allows us to easily define new FF anyware. But as a price, our FF are going to be spreaded all over the project!

Reply:
technically you could do that before as well.
we could think of unexporting the featureflag type (and not introducing New)
this way we force the developer to introduce feature flags only via exported funcs..